manifests/record/mx.pp

```diff
@@ -1,15 +1,36 @@
 # == Define: dns::record::mx
 #
-# Wrapper for dns::record to set an XM record.
+# Wrapper for dns::record to set an MX record.
 #
 define dns::record::mx (
   $zone,
   $data,
-  $ttl = '',
-  $preference = '0',
-  $host = $name ) {
+  $ttl        = '',
+  $preference = 10,
+  $host       = '@' ) {
 
-  $alias = "${host},MX,${zone}"
+  $alias = "${host},${zone},MX,${preference},${data}"
+
+  unless is_string($zone) and is_domain_name($zone) and !is_ip_address($zone) and $zone !~ /\.?\d+$/ {
```

Comment:
Hmm. That's fine. I'm mostly worried about the failing test and would be "ok" to me if you dropped to the strictness to make it pass. Or just figure out why it is failing. (maybe it doesn't like the "!"?)